### PR TITLE
Sundry fixes for CNF quickstart

### PR DESCRIPTION
**Pull request by sunnycarter**
* Various fixes to NFD build of deployParameters schema and interactive mode create of deployParameters mappings file.
* Fix CNF NFD publish so that it doesn't render the ACR unuseable for future Artifact publishing.
* Allow CNF NFD image copy from a source ACR using a namespace.
* Fix - Add new CGSchema parameters not from the NFD to the `required` section of the schema.
* Add the ability to skip bicep publish or artifact upload during publish commands.
* Fix Manifest name for NSDs so it isn't the same as that for NFDs
* Add validation of source_registry_id format for CNF configuration

Required for https://dev.azure.com/msazuredev/AzureForOperators/_git/pez-networkservicedesigns/pullrequest/62630 to merge

Sadly, git hasn't worked out that quite a lot of changes are just indenting big blocks of stuff. Because of merge conflicts with Andy's PR I had to manually fix up a lot here and now it doesn't know that it is just an indentation change. sorry :( That's all in deploy_with_arm.py.

### CODE REVIEW

File: src/aosm/azext_aosm/generate_nsd/nsd_generator.py [hunk 1]
```diff
@@ -130,27 +130,40 @@ def config_group_schema_dict(self) -> Dict[str, Any]:
...
             "type": "string",
             "description": managed_identity_description_string,
         }
+        cgs_dict["required"].append("managedIdentity")
+
+        if self.config.network_function_type == CNF:
+            custom_location_description_string = (
+                "The custom location ID of the ARC-Enabled AKS Cluster to deploy the CNF "
+                "to. Should be of the form "
+                "'/subscriptions/c7bd9d96-70dd-4f61-af56-6e0abd8d80b5/resourcegroups"
```
Comment (jamiedparsons) [inline src/aosm/azext_aosm/generate_nsd/nsd_generator.py:160]: Can you remove the subscription ID from the help text?

File: src/aosm/azext_aosm/_params.py [hunk 2]
```diff
@@ -109,6 +110,9 @@ def load_arguments(self: AzCommandsLoader, _):
                 " alternative parameters."
             ),
         )
+        c.argument(
+            "skip", arg_type=skip_steps, help="Optional skip steps"
```
[the remark this answers is not in the document]
Comment (sunnycarter) [reply src/aosm/azext_aosm/_params.py:114]: Yes, something in the CLI does this for us. You get 
`--skip                         : Optional skip steps.  Allowed values: artifact-upload, bicep-
                                     publish.`

File: src/aosm/azext_aosm/deploy/deploy_with_arm.py [hunk 3]
```diff
@@ -106,50 +111,59 @@ def deploy_vnfd_from_bicep(
...
-        )
-        print(message)
-        logger.info(message)
-        self.deploy_bicep_template(bicep_path, parameters)
-        print(f"Deployed NFD {self.config.nf_name} version {self.config.version}.")
+            print("Skipping bicep publish")
+
+        if skip == ARTIFACT_UPLOAD:
+            print("Skipping artifact upload")
+            print("Done")
```
[the remark this answers is not in the document]
Comment (sunnycarter) [reply src/aosm/azext_aosm/deploy/deploy_with_arm.py:165]: I like it.  If you don't skip you get Done at the end of everything so I like having it here too.

File: src/aosm/azext_aosm/deploy/pre_deploy.py [hunk 4]
```diff
@@ -140,23 +147,25 @@ def ensure_config_source_registry_exists(self) -> None:
...
-        self.api_clients.container_registry_client.get(
-            resource_group_name=source_registry_resource_group_name,
-            registry_name=source_registry_name,
-        )
+        # Match the source registry format
+        source_registry_match = re.search(SOURCE_ACR_REGEX, self.config.source_registry_id)
+        # Config validation has already checked and raised an error if the regex doesn't
+        # match
+        if source_registry_match and len(source_registry_match.groups()) > 1:
+            source_registry_resource_group_name = source_registry_match.group(1)
```
[the remark this answers is not in the document]
Comment (sunnycarter) [reply src/aosm/azext_aosm/deploy/pre_deploy.py:161]: I didn't know about these, thanks.

**Pull request merged**